synapseclient/models/team.py

```diff
@@ -356,3 +431,28 @@ async def open_invitations_async(
             team=self.id, synapse_client=synapse_client
         )
         return list(invitations)
+
+    async def get_user_membership_status_async(
+        self,
+        user_id: str,
+        *,
+        synapse_client: Optional[Synapse] = None,
+    ) -> TeamMembershipStatus:
+        """Retrieve a user's Team Membership Status bundle for this team.
+
+        Arguments:
+            user_id: Synapse user ID
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Returns:
+            TeamMembershipStatus object
+        """
```

Comment:
Could you add in an example of using this function (And one in the non-async version)?